.github/workflows/main.yaml

```diff
@@ -0,0 +1,139 @@
+name: CI Workflow
+
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+      - develop
+  workflow_dispatch:
+
+env:
+  COMPONENT_BUILDER: builder
+  COMPONENT_API: api
+  AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+  AWS_ECR_REGION: us-east-2
+
+jobs:
+
+  lint-dockerfile:
+    runs-on: ubuntu-latest
```

Comment:
do you need ubuntu-latest? or shall we fixed to a certain LTS version so that it would be easier to maintain.